queue_job/job.py

```diff
@@ -397,20 +405,33 @@ def store(self):
         if self.eta:
             vals['eta'] = dt_to_string(self.eta)
 
-        db_record = self.db_record()
+        db_record = self.db_record(use_new_cursor=use_new_cursor)
         if db_record:
-            db_record.write(vals)
+            if use_new_cursor:
+                cr = registry(db_record.env.cr.dbname).cursor()
+                db_record.with_env(db_record.env(cr=cr)).write(vals)
+                cr.commit()
+                cr.close()
+            else:
+                db_record.write(vals)
         else:
             date_created = dt_to_string(self.date_created)
             vals.update({'uuid': self.uuid,
                          'name': self.description,
                          'date_created': date_created,
                          })
-
-            self.job_model.sudo().create(vals)
-
-    def db_record(self):
-        return self.db_record_from_uuid(self.env, self.uuid)
+            if use_new_cursor:
+                cr = registry(self.job_model.env.cr.dbname).cursor()
+                self.job_model.with_env(
+                    self.job_model.db_record.env(cr=cr)).sudo().create(vals)
+                cr.commit()
+                cr.close()
```

Comment:
The close should be in a `finally`.
You could factor this out with a context manager

```python

@contextmanager
def use_new_env(self, env):
    cr = registry(env).cursor()
    new_env = env(cr=cr)
    try:
        yield new_env
    except:
        cr.rollback()
        raise
    else:
        cr.commit()
    finally:
        cr.close()